### ISSUE TITLE
[java] TestClassWithoutTestCases false positive with JUnit5 ParameterizedTest

### ISSUE THREAD

**Issue by andreoss** [pmd/pmd#2636]
**Affects PMD Version:** 6.10.0

**Rule:** [TestClassWithoutTestCases](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#testclasswithouttestcases)

**Description:**

When a test class is extended from some parent class, the  following check is triggered: 
**This class name ends with Test but contains no test cases (TestClassWithoutTestCases)**
Probably its being misidentified as Junit 3.

```java
final class SampleTest extends Object {

    @ParameterizedTest
    @ValueSource(ints = {1})
    void doesTest(final int num) {
        Assertions.assertTrue(num > 0);
    }
}
```

Test class which has no parent works fine, but empty class is not producing an error too (probably worth a separate issue).

```java
final class SampleTest {
}
```

**Expected outcome:**

No error occurs when `@ParameterizedTest` is used inside a test class with a parent.

**Comment by LiGaOg** [comment]
Please assign this issue to me

**Comment by BillyBolton** [comment]
Any status on this issue?

**Comment by adangel** [comment]
I couldn't reproduce this false-positive when using the `auxclasspath` option (typeresolution). Only when running PMD on an non-compiled project without auxclasspath, this violation was reported. And with the current version (PMD 6.50.0, actually since PMD 6.27.0) it is not reported even without auxclasspath.

However, looking at the implementation of TestClassWithoutTestCases - the impl doesn't support "ParameterizedTest". So there are two problems:

1. ParameterizedTest is not supported (and some other annotations from JUnit5)
2. Empty classes are not detected - this is a false negative, see the older issue #929 

I'm trying to fix both issues.
